### PR TITLE
MRG: Algorithm PETS

### LINKED ISSUE #35: Implement PETS
* Paper: Deep Reinforcement Learning in a Handful of Trials using Probabilistic Dynamics Models by Chua et al., https://arxiv.org/pdf/1805.12114
* Papers with code: https://paperswithcode.com/paper/deep-reinforcement-learning-in-a-handful-of
* Official code (using TensorFlow 1.9): https://github.com/kchua/handful-of-trials
* Reimplementation by main author (using JAX): https://github.com/kchua/mbrl-jax
* Facebook's implementation (using PyTorch): https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/algorithms/pets.py
* Implementation by some guy (using PyTorch): https://github.com/quanvuong/handful-of-trials-pytorch
* Implementation of probabilistic network ensemble (using PyTorch): https://github.com/johannesnauta/pytorch-pne
* Another implementation (using PyTorch): https://github.com/michaelbayerle/PETS

### PR DESCRIPTION
**Pull request by AlexanderFabisch**
Fix #35 

## TODO list
- [x] JAX reimplementation of a Gaussian MLP
- [x] JAX reimplementation of ensemble for epistemic uncertainty estimation
- [x] CEM
- [x] train planning model
- [x] MPC with CEM on model
- [x] full PETS
- [x] speed up
- [x] fix nans in dynamic model
    - [x] output log_vars
    - [x] make boundaries trainable
        - https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py#L150
        - https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py#L304
        - init: https://github.com/kchua/handful-of-trials/blob/master/dmbrl/modeling/models/BNN.py#L159
        - loss: https://github.com/kchua/handful-of-trials/blob/master/dmbrl/modeling/models/BNN.py#L182
        - use: https://github.com/kchua/handful-of-trials/blob/master/dmbrl/modeling/models/BNN.py#L414
    - [x] how is this compatible to vmapped TrainStates? -> check jax docs; solution: nnx interface
    - [x] mini-batch training
    - [x] clean up, refactor
- [ ] configure full training pipeline similar to original implementation
    - [ ] https://github.com/kchua/handful-of-trials/blob/master/dmbrl/misc/MBExp.py#L93
    - [ ] https://github.com/kchua/handful-of-trials/blob/master/dmbrl/controllers/MPC.py#L160
    - [ ] https://github.com/kchua/handful-of-trials/blob/master/dmbrl/modeling/models/BNN.py#L220
    - [x] L2 regularization, common weight 0.0005 or 0.00075, [example config](https://github.com/kchua/handful-of-trials/blob/77fd8802cc30b7683f0227c90527b5414c0df34c/dmbrl/config/reacher.py#L78), [loss](https://github.com/kchua/handful-of-trials/blob/master/dmbrl/modeling/layers/FC.py#L127)
- [x] clean up
- [x] properly document code

## Result

After 3 episodes of random sampling and training the dynamics model, this is the result of the MPC policy (showing 3 episodes with random initial state, speed x20):

https://github.com/user-attachments/assets/0254b682-5f46-4921-9cae-b3147a08cbc5

## Review TODO List

- [ ] Create a function for MPC

### CODE REVIEW

**Review by mlaux1 — DISMISSED** [review]
Looks good

**Comment by AlexanderFabisch** [comment]
Solved the merge conflict, should I merge it then?

**Comment by mlaux1** [comment]
> Solved the merge conflict, should I merge it then?

Yes, please!
